sql/core/src/test/resources/sql-tests/inputs/cast.sql

```diff
@@ -42,4 +42,38 @@ SELECT CAST('9223372036854775808' AS long);
 
 DESC FUNCTION boolean;
 DESC FUNCTION EXTENDED boolean;
+
+-- cast null to calendar interval should return null
+SELECT CAST(NULL as calendarinterval);
+SELECT CALENDARINTERVAL(NULL);
+
+-- cast invalid strings to calendar interval should return null
+SELECT CAST('interval 10' as calendarinterval);
+SELECT CAST('interval 100 nanoseconds' as calendarinterval);
+SELECT CAST('interval 1 second 10 years -10 months 1 minute' as calendarinterval);
+SELECT CAST('interval 60 hours + 1 minute' as calendarinterval);
+SELECT CAST('interval 1 day +5 minutes' as calendarinterval);
+
+-- cast valid strings to calendar interval should return calendar interval
+SELECT CAST('interval 5 minutes' as calendarinterval);
+SELECT CAST('interval 10 hours' as calendarinterval);
+SELECT CAST('interval 1 second' as calendarinterval);
+SELECT CAST('interval 3 years -3 month 7 week 123 microseconds' as calendarinterval);
+SELECT CAST('interval 100 years 15 months -24 weeks 66 seconds' as calendarinterval);
+
+-- casting invalid strings to calendar interval using the function should return null
```

Comment:
i wouldn't add this many test cases, since they are just simple aliases. jsut add one for calendarinterval and keep the cast ones.